src/Services/Catalog/Catalog.API/Controllers/CatalogController.cs

```diff
@@ -38,48 +42,24 @@ public CatalogController(CatalogContext context, IOptionsSnapshot<CatalogSetting
         [ProducesResponseType(typeof(IEnumerable<CatalogItem>), (int)HttpStatusCode.OK)]
         public async Task<IActionResult> Items([FromQuery]int pageSize = 10, [FromQuery]int pageIndex = 0, [FromQuery] string ids = null)
         {
-            if (!string.IsNullOrEmpty(ids))
-            {
-                return GetItemsByIds(ids);
-            }
-
-            var totalItems = await _catalogContext.CatalogItems
-                .LongCountAsync();
+            var cacheFilter = new CacheCatalogFilter { Key = $"{Request.Path.Value}{Request.QueryString}", PageSize = pageSize, PageIndex = pageIndex };
```

Comment:
my concern here is if the pagesize is page 1 and page index is 10 or if the page is 11 and page index is 0 we ended up pulling the wrong values. We need to correct this.

Reply:
the way we are storing the key can cause issues
example:
Request.Path.Value="randomvalue";
Request.QueryString="/path/to/product/";

pageSize =101;
PageIndex=0;
 Key = $"{Request.Path.Value}{Request.QueryString}", PageSize = pageSize, PageIndex = pageIndex };
Save.Cache(Key); //key is = "randomvalue/path/to/product

pageSize =10;
PageIndex= 10;
 Key = $"{Request.Path.Value}{Request.QueryString}", PageSize = pageSize, PageIndex = pageIndex };  //key is = "randomvalue/path/to/product

Save.Cache(Key);

// we now have duplication issues at the key level creating inconsistent data; lets separate out the   data using some sort of pipelines so data is well defined at the key level and including pagesize and page index as part of the caching key name.
Key = $"{Request.Path.Value}||{Request.QueryString}||pageSize||pageIndex"';

Reply:
Reviewed with group,   Since  Query is coming fromQuery and its already present its a non issue.